app/shop.py

```diff
@@ -0,0 +1,5 @@
+class Shop:
+    def __init__(self, shop_info: dict) -> None:
+        self.name = shop_info["name"]
+        self.location = shop_info["location"]
+        self.products = shop_info["products"]
```

Comment:
use get() method instead of parenthesis everywhere for getting value by key to avoid KeyError